redash/handlers/data_sources.py

```diff
@@ -182,19 +187,9 @@ def get(self, data_source_id):
         require_access(data_source, self.current_user, view_only)
         refresh = request.args.get("refresh") is not None
 
-        response = {}
-
-        try:
-            response["schema"] = data_source.get_schema(refresh)
-        except NotSupported:
-            response["error"] = {
-                "code": 1,
-                "message": "Data source type does not support retrieving schema",
-            }
-        except Exception:
-            response["error"] = {"code": 2, "message": "Error retrieving schema."}
```

Comment:
We need to keep reporting these error codes. For example, we might show different UI when it's not supported. And we don't want to throw random errors at the user but rather show "Error retrieving schema." (it might make sense to show some of the errors, but it's beyond the scope of this PR).

Reply:
I thought so too, but then I realized we aren't really using the [error codes defined in the service](https://github.com/getredash/redash/pull/4797/commits/75a48b098082385ab5241b641239a115e19bed65#diff-60f072516fcb2847feb283641b82e60bL4). The current implementation (returning `[]` on `NotSupported` from the backend) mimics the same behavior we have today - silent failing on `NotSupported` and a "Schema refresh failed." message when errors occur.

Reply:
I know that we don't really use them anymore in code -- this is something that happened during the conversion to React, but we should someday. It's better to be explicit about this.

Reply:
bc69c032a